### PR TITLE
Added role field to Process

### PR DESCRIPTION
**Pull request by daveirwin1**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by daveirwin1** [comment]
I'm not sure why it shows as 9 commits.  There should be only one.  This file already existed.  We just added the role field to the Process response type.

File: arm-service-map/2015-11-01-preview/swagger/arm-service-map.json [hunk 1]
```diff
@@ -0,0 +1,2929 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/features/serviceMap/machines": {
```
Comment (dsgouda) [inline arm-service-map/2015-11-01-preview/swagger/arm-service-map.json]: @ravbhatnagar I'm assuming we should block on this requirement but please confirm

**Review by dsgouda — COMMENTED** [review]
Partial review, will post more comments soon

File: arm-service-map/2015-11-01-preview/swagger/arm-service-map.json (hunk 1 above)
Comment (daveirwin1) [inline arm-service-map/2015-11-01-preview/swagger/arm-service-map.json]: This part of the spec did not change.  We've already been through review with these paths, the API has been published, and customers have written code to this spec.

File: arm-service-map/2015-11-01-preview/swagger/arm-service-map.json [hunk 2]
```diff
@@ -0,0 +1,2929 @@
...
+            }
+          }
+        }
+      }
+    },
+    "ClientGroup": {
+      "description": "Represents a collection of clients of a resource. A client group can represent the clients of a port, process, or a machine.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CoreResource"
```
Comment (daveirwin1) [inline arm-service-map/2015-11-01-preview/swagger/arm-service-map.json]: I think maybe this is the wrong code segment to go with the comment about duration.  I agree that it's always better to be more clear.  However, this is not a part of the spec that changed in this PR.

File: arm-service-map/2015-11-01-preview/swagger/arm-service-map.json [hunk 3]
```diff
@@ -0,0 +1,2929 @@
...
+              "type": "string",
+              "enum": [
+                "webServer",
+                "appServer",
+                "databaseServer",
+                "ldapServer",
+                "smbServer"
+              ],
+              "description": "The inferred role of this process based on its name, command line, etc."
+            },
```
Comment (daveirwin1) [inline arm-service-map/2015-11-01-preview/swagger/arm-service-map.json]: This is the only part of the file that changed.  We added the "role" section.  I don't understand why GitHub seems to think that this entire file is new.

**Comment by dsgouda** [comment]
@daveirwin1 Try rebasing the commits, or recreating the branch with just the changes you have made

**Comment by daveirwin1** [comment]
Closing, because there's confusion over what changed here.  For some reason, Git is including revisions that had already been committed and merged.

**Pull request closed without merging**